DisguiseUnityRenderStream/Runtime/DisguiseClusterDisplay.cs

```diff
@@ -69,83 +118,143 @@ public static void RegisterClusterDisplayEvents()
+                // of node ids in use.
+                var retries = 0;
+                var nodeIdFilePath = Path.Combine(Path.GetTempPath(), Application.productName + ".node");
+                while (clusterParams.NodeID == 255 && retries < 10)
+                {
+                    try
                     {
-                        newFrameData = false;
-                        SceneManager.LoadScene((int)frameData.scene);
+                        using var fileStream = new FileStream(nodeIdFilePath, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.None);
+                        using var reader = new StreamReader(fileStream);
+                        if (!byte.TryParse(reader.ReadLine(), out clusterParams.NodeID))
+                        {
+                            clusterParams.NodeID = 0;
+                        }
+
+                        fileStream.Position = 0;
+                        using var writer = new StreamWriter(fileStream);
+                        writer.Write(clusterParams.NodeID + 1);
+                        fileStream.SetLength(fileStream.Position);
                     }
+                    catch (Exception ex)
+                    {
+                        retries++;
+                        ClusterDebug.Log($"Unable to access node ID file: {ex.Message}");
+                        Thread.Sleep(500);
+                    }
+                }
 
-                    break;
+                Application.quitting += () =>
+                {
+                    File.Delete(nodeIdFilePath);
+                };
+
+#if NET_4_6
             }
+#endif
+            ClusterDebug.Log($"Auto-assigning node ID {clusterParams.NodeID} (repeaters: {clusterParams.RepeaterCount})");
+            // First one to start up gets to be the emitter - node 0
+            clusterParams.EmitterSpecified = clusterParams.NodeID == 0;
 
-            newFrameData = (error == RS_ERROR.RS_ERROR_SUCCESS);
-        
-            if (newFrameData)
+            if (clusterParams.NodeID == 255 || clusterParams.RepeaterCount == 0)
+            {
+                ClusterDebug.LogWarning("Cannot obtain cluster settings. Cluster rendering will be disabled");
+                clusterParams.ClusterLogicSpecified = false;
+            }
+
+            return clusterParams;
+        }
+
+        public void Dispose()
+        {
+            ClusterSyncLooper.onInstanceDoPreFrame -= AwaitFrame;
+            ClusterSyncLooper.onInstanceDoPreFrame -= PublishEmitterEvents;
+            ClusterSyncLooper.onInstanceDoLateFrame -= PublishEmitterEvents;
+            m_FollowerFrameDataSubscription?.Dispose();
+            m_FrameDataBus?.Dispose();
+            m_FrameDataBus = null;
+            if (m_ClusterSync != null)
             {
-                ProcessFrameData(frameData);
+                m_ClusterSync.DisableClusterDisplay();
+                ServiceLocator.Withdraw<IClusterSyncState>();
```

Comment:
Minor: We seem to shield ourselves from double dispose for m_FrameDataBus (setting it to null).  Should we also do it here for m_ClusterSync?